scripts/migration/migrate-addon.sh

```diff
@@ -0,0 +1,653 @@
+    prompt_unsupported "Detected possibly incompatible virtualservice config" "websocketUpgrade attribute is no longer supported"
+  fi
+  if grep -q "abort\.percent" <<< "${VIRTUALSERVICES}" ; then
+    prompt_unsupported "Detected possibly incompatible virtualservice config" "abort.percent attribute is no longer supported"
+  fi
+
+  # Custom gateway
+
+  # Check custom mixer policies
+  check_custom_mixer
+  LAST_ERR=$?
+  if [[ "${LAST_ERR}" != "0" ]]; then
+    prompt_unsupported "Detected custom mixer resources that could not be migrated" "requries manual migration to telemetryv2"
+  fi
+}
+
+print_restart_instructions() {
+  echo
+  echo "The migration completed successfully. Now, please perform the following steps:"
+  echo
+  echo "1. Restart all workloads in Istio-enabled namespaces to update the sidecar proxies to the new version."
+  echo
+  echo "2. Any manually injected proxies must also be re-injected manually at this point."
+}
+
+cleanup() {
+  # TODO: ensure migration is complete, nothing is using addon, and the addon is not installed before proceeding
+  heading "Cleaning up old resources..."
+  STATE="$(kube get cm -n istio-system asm-addon-migration-state -ojsonpath='{.data.migrationStatus}' || true)"
+  if [[ "${STATE}" != "COMPLETE" ]]; then
+    die "Migration must be completed before --cleanup can run"
+  fi
+  found=()
+  for rs in "${CLEANUP_RESOURCES[@]}"; do
+    kind="$(echo "${rs}" | cut -d/ -f1)"
+    name="$(echo "${rs}" | cut -d/ -f2)"
+    ns="$(echo "${rs}" | cut -d/ -f3)"
+    if [[ "${ns}" == "" ]]; then
+      RESP="$(kubectl --context="${CONTEXT}" get "${kind}" "${name}" --ignore-not-found -oname)"
+    else
+      RESP="$(kubectl --context="${CONTEXT}" get "${kind}" "${name}" -n "${ns}" --ignore-not-found -oname)"
+    fi
+    if [[ "${RESP}" != "" ]]; then
+      echo "Will delete ${kind}/${name}.${ns}"
+      found+=("${rs}")
+    fi
+  done
+  prompt "Deleting resources listed above..."
+  for rs in "${found[@]}"; do
+    kind="$(echo "${rs}" | cut -d/ -f1)"
+    name="$(echo "${rs}" | cut -d/ -f2)"
+    ns="$(echo "${rs}" | cut -d/ -f3)"
+    if [[ "${ns}" == "" ]]; then
+      kube delete "${kind}" "${name}" --ignore-not-found
+    else
+      kube delete "${kind}" "${name}" -n "${ns}" --ignore-not-found
+    fi
+  done
+  kube delete -n istio-system jobs -lk8s-app=istio,app=security
+}
+
+fn_exists() { declare -F "$1" > /dev/null; }
+
+arg_required() {
+  if [[ ! "${2:-}" ]]; then
+    die "Option ${1} needs an argument"
+  fi
+}
+
+main_run() {
+  check_prerequisites
+  print_intro
+  config_check
+  disable_galley_webhook
+  migrate_configs
+  if [[ -n $ZERO_DOWNTIME ]]; then
+    configure_mesh_ca
+  fi
+  replace_gateway
+  write_marker "COMPLETE"
+  # replace_webook and cleanup are invoked seperately and needed for completion
+  print_restart_instructions
+}
+
+usage() {
+    echo "Usage:"
+    echo "  $0 [OPTIONS] [--command] [command-name]"
```

Comment:
Let's make it more obvious that if you leave out `--command` it runs all of them.

Reply:
The tool now requires an explicit command for it to "run everything". This command is hidden from usages as it is not meant to be used in interactive mode